Tools/Scripts/webkitpy/w3c/test_exporter.py

```diff
@@ -313,10 +377,24 @@ def format(self, record):
 
 
 def main(_argv, _stdout, _stderr):
-    options = parse_args(_argv)
-
     configure_logging()
+    host = Host()
 
-    test_exporter = TestExporter(Host(), options)
+    if not has_wpt_test_changes(_argv, host):
+        _log.info('No changes to upstream. Exiting...')
+        return
 
+    export_wpt_test_changes(_argv, host)
+
+
+def export_wpt_test_changes(args, host):
+    options = parse_args(args)
+    wpt_patch_generator = WebPlatformTestPatchGenerator(host, options)
+    test_exporter = TestExporter(host, options, wpt_patch_generator)
     test_exporter.do_export()
+
+
+def has_wpt_test_changes(args, host):
+    options = parse_args(args)
+    wpt_patch_generator = WebPlatformTestPatchGenerator(host, options)
+    return wpt_patch_generator.has_wpt_changes()
```

Comment:
Looking for some feedback on how I can refactor this so `wpt_patch_generator` isn't instantiated twice. `has_wpt_test_changes` and `export_wpt_test_changes` are both used by `Tools/Scripts/webkitpy/tool/steps/wptchangeexport.py` to determine if it should ask the user if they want to export the changes.

Reply:
Spoke in person - collective conculsion was to merge the 2 classes together and adjust the prompts so they appear at the right time in the new workflow.